src/sw/sw.ts

```diff
@@ -33,20 +33,90 @@ async function pingAnalytics(eventAction: string) {
 }
 
 type CustomNotification = {
-  data: NotificationURLData|void;
+  data: NotificationData|void; close: () => Promise<void>;
 };
 
 // This is displayed iff there is no data in the payload
 const DEFAULT_NOTIFICATION_OPTIONS: NotificationPayload = {
   title: 'New Github Health Updates',
   body: 'Click the notification to view updates',
+  icon: '/images/notification-images/icon-192x192.png',
+  badge: '/images/notification-images/badge-128x128.png',
   requireInteraction: false,
   data: {
     url: new URL('/', self.location.origin).toString(),
   },
   tag: 'default-notification',
 };
 
+async function cleanupNotifications(notifications: CustomNotification[]):
+    Promise<void> {
+  try {
+    // Using set to remove duplicates
+    const openPullRequests = new Set();
+    // This map allows us to link notifications to a specific PR
+    const prsToNotifications: {[key: string]: CustomNotification[]} = {};
+
+    for (let i = 0; i < notifications.length; i++) {
+      const notification: CustomNotification =
+          // tslint:disable-next-line:no-any
+          (notifications[i] as any) as CustomNotification;
+      if (!notification.data) {
+        continue;
+      }
+
+      const data: NotificationData = notification.data;
+      if (data.pullRequest) {
+        openPullRequests.add(data.pullRequest.gqlId);
+
+        if (!prsToNotifications[data.pullRequest.gqlId]) {
+          prsToNotifications[data.pullRequest.gqlId] = [];
+        }
+        prsToNotifications[data.pullRequest.gqlId].push(notification);
+      }
+    }
+
+    // Convert the set (used to remove duplicates) and create an array
+    // of PR's to check state of.
+    const pullRequestDetails = Array.from(openPullRequests);
+
+    // If there are no pull requests, don't bother making a network
+    // request.
+    if (pullRequestDetails.length === 0) {
+      return;
+    }
+
+    const response = await fetch('/api/check-pr-state', {
+      method: 'POST',
+      credentials: 'include',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(pullRequestDetails),
```

Comment:
So this looks like `['abc', 'id2']` right?

Reply:
Yes it will.